pallets/schemas/src/lib.rs

```diff
@@ -243,10 +251,9 @@ pub mod pallet {
 		}
 
 		pub fn get_schema_by_id(schema_id: SchemaId) -> Option<SchemaResponse> {
-			// TODO: This will eventually be a struct with other properties. Currently it is just the model
-			if let Some(model) = Self::get_schema(schema_id) {
+			if let Some(schema) = Self::get_schema(schema_id) {
 				// this should get a BoundedVec out
-				let model_vec = model.into_inner();
+				let model_vec = schema.model.into_inner();
 				let response = SchemaResponse { schema_id, model: model_vec };
```

Comment:
I think `SchemaResponse` should also return the type

Reply:
It is `Option<SchemaResponse>` because there might not be a schema with that id.

Reply:
You still need to assign mode_type from schema to SchemaResponse